bacpac v2

mgeaghan opened the pull request:
Large update:

- Removed trycycler
- Refactored assembly into a `denovo` subworkflow
    - Used by both main workflow as well as `autocycler` subworkflow
- Added plassembler outputs into autocycler so we include plasmids in final consensus assembly
- For samples where one of the de novo assemblies is used instead of the consensus, both the 'best' de novo assembly and the corresponding plassembler assembly will be processed by the AMR and virulence gene detection processes
    - In these cases, the results from the de novo chromosome and plasmid assemblies will be combined when generating the final heatmap
- Created new process to generate final heatmap using altered version of previous R script.
    - Also added some bash code to the end of the amrfinder and abricate processes to export a second annotated TSV file. Includes sample name and assembler so that all of these files from all samples can be concatenated and simplify the process of generating a heatmap.

mgeaghan committed:
WIP: remove trycycler, refactor with denovo subworkflow

mgeaghan committed:
WIP: update annotation stage, clean up main script

mgeaghan committed:
remove all references to multiqc config params

mgeaghan committed:
wrap seq summary in frompath method

mgeaghan committed:
remove params references in process scripts

mgeaghan committed:
create path channel for kraken db

mgeaghan committed:
create dataflow values from single file channels

mgeaghan committed:
allow all plassembler outputs to be optional

mgeaghan committed:
continued: allow all plassembler outputs to be optional

mgeaghan committed:
remove subset from plassembler fasta channel

mgeaghan committed:
add conditional assembly path to medaka to handle different plassembler name

mgeaghan committed:
add plassembler fasta name check to autocycler compress

mgeaghan committed:
replace old amr phylo summary code

mgeaghan committed:
fix column to rowname call

mgeaghan committed:
remove duplicated header lines from amr reports

mgeaghan committed:
clean up heatmap code

mgeaghan committed:
remove old commented out code

mgeaghan committed:
remove barcode reference in autocycler table mqc process

mgeaghan committed:
put plassembler check into bash script for autocycler compress

mgeaghan commented:
Update: I found that the actual source of the bug in the heatmap code is the `cbind` call that is used to combine the abricate and amrfinder matrices together. This is a very naive function that doesn't compare row names, so if the two dataframes have different row orders, the labels for the second dataframe (i.e. the abricate data) will be wrong (but the first dataframe labels will be correct - in this case the amrfinder outputs).

Unfortunately, this `cbind` call is still in my code. I will fix it now and push the new update.

mgeaghan committed:
heatmap: remove old cbind method and replace with full join

mgeaghan committed:
delete old files

mgeaghan committed:
refactor phylo file script, add sample info file to output of bakta

mgeaghan committed:
delete more unused files, use regex to detect assemblers and sample names

mgeaghan committed:
fix input type for create phylo tree files

mgeaghan committed:
fix path issues in create phylo files script

mgeaghan committed:
update readmes

mgeaghan committed:
update flow chart

Mitchob commented:
First test on barcodes 04 & 05. The pipeline ran smoothly with no execution issues.

- I got the following warnings I think they're related to processes referenced in nextflow.config but not present in the workflow:
```
WARN: There's no process matching config selector: check_input
WARN: There's no process matching config selector: check_samplesheet
WARN: There's no process matching config selector: flye_assembly_subset
WARN: There's no process matching config selector: unicycler_assembly_subset
WARN: There's no process matching config selector: trycycler_subsample -- Did you mean: autocycler_subsample?
WARN: There's no process matching config selector: trycycler_cluster -- Did you mean: autocycler_cluster?
WARN: There's no process matching config selector: trycycler_cluster_subset
WARN: There's no process matching config selector: trycycler_reconcile
WARN: There's no process matching config selector: trycycler_msa
WARN: There's no process matching config selector: trycycler_partition
WARN: There's no process matching config selector: trycycler_consensus
```
- Should we consider renaming the final MultiQC output file from `multiqc_report.html` to something more descriptive like `multiqc_report_analysis.html` and `multiqc_report_quality.html`? Could help with file organisation and conflicts when downloading multiple reports.
- Should we include if89 database paths for Gadi under /gadi-execution.md? This could help users run the pipeline more easily without having to manually locate database directories.

Currently rerunning on additional barcodes, the initial tests only included one species and no plasmids. I haven't gone through the documentation yet to see what aspects need to be updated for v2.

Mitchob commented:
I added barcodes 3 and 6 and the pipeline ran smoothly without any issues. It correctly used cached data for previously processed samples. 

feeback on MultiQC Reports

For both I think It would be helpful to add brief interpretation text for each output section. Some of this may already exist in the interpretation markdown and could be reused.

Quality Report
- Could the General Stats → pycoqc_summary.json entry be moved into the pycoQC section for consistency?
- For Autocycler metrics, can the barcode metrics and barcode cluster metrics be split out into separate tables?
- In autocycler metrics there are three columns with no output (input_read_count, input_read_bases, input_read_n50). Are these expected to be empty and can they be removed?
- For the Bandage output, lets include a note that node colours are random (unless they convey a defined meaning that I am missing. In which case, please clarify).

Results Report
- In General Stats, can the % species metrics be separated from the assembly statistics?
- What additional value does “% species” provide beyond what is already shown in the “Kraken top taxa” section? Is this a commonly reported metric (e.g., top 5%)?
- In the barcodes I ran 03-06, it’s unclear why Vibrio campbellii is not also captured by the % species metrics in General stats?
-Heatmaps look great.  I think a present absent key (or note) for the colours will help with interpretation.

Mitchob reviewed (approved):
I’ve added a few suggestions, mostly nice-to-haves, as the pipeline is running smoothly overall.
Just checking: have you had a chance to review the documentation and remove any references to Trycycler and include any updates? I havent looked yet

mgeaghan committed:
remove old comments, update param defs and schema

mgeaghan commented:
@Mitchob check out my latest commit [c817653](https://github.com/Sydney-Informatics-Hub/ONT-bacpac-nf/pull/113/commits/c817653deb5d57357bf1edd8275468ed51765d23). I've updated the schema and removed some old comments. Also update the default values for some params, e.g. input_directory and samplesheet are now empty strings by default, not false; still falsy value for checking which input method to use, so no issues there, but shouldn't mess with seqera anymore.

mgeaghan committed:
add assembly options and other sections to allof
